backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_PriceList_Details_ToDate_Report.sql

```diff
@@ -68,17 +76,32 @@ BEGIN
     CREATE TEMP TABLE IF NOT EXISTS temp_price_list_details AS
     SELECT DISTINCT bp.c_bpartner_id,
                     plv.M_PriceList_Version_ID,
-                    plv.name
+                    plv.name,
+                    (SELECT bpl.c_bpartner_location_id
+                     FROM c_bpartner_location bpl
+                     WHERE bpl.c_bpartner_id = bp.c_bpartner_id
+                     ORDER BY bpl.isbilltodefault
+                             DESC
+                     LIMIT 1) AS c_bpartner_location_id,
+                    bp.ad_org_id
     FROM c_bpartner bp
              INNER JOIN M_PriceList pl ON bp.m_pricingsystem_id = pl.m_pricingsystem_id
              INNER JOIN M_PriceList_Version plv ON pl.m_pricelist_id = plv.m_pricelist_id
     WHERE bp.iscustomer = 'Y'
       AND pl.issopricelist = 'Y'
-      AND plv.validfrom >= p_validfrom;
+      AND pl.issopricelist = p_issotrx
+      AND plv.validfrom >= p_validfrom
+      AND (p_C_BPartner_ID IS NULL OR bp.c_bpartner_id = p_C_BPartner_ID)
+      AND (p_C_BP_Group_ID IS NULL OR bp.c_bpartner_id IN (SELECT DISTINCT b.c_bpartner_id
+                                                           FROM c_bpartner b
```

Comment:
Why writing 
```SQL
  AND (p_C_BP_Group_ID IS NULL OR bp.c_bpartner_id IN (SELECT DISTINCT b.c_bpartner_id
                                                           FROM c_bpartner b
```
Instead of  
```SQL
AND (p_C_BP_Group_ID IS NULL OR bp.C_BP_Group_ID = p_C_BP_Group_ID )
```

Reply:
done